cypress-io/cypress
# fix: force gzip when no accept encoding header is sent and use identity if gzip is not sent

- Closes #28025 

### Additional details


Currently, Cypress [tries to filter down](https://github.com/cypress-io/cypress/blob/develop/packages/proxy/lib/http/request-middleware.ts#L348-L374) Accept-Encoding headers to just gzip. This works when the header is present, but when it is blank according to the [standards for the header](https://www.rfc-editor.org/rfc/rfc9110#field.accept-encoding):

> If no Accept-Encoding header field is in the request, any content coding is considered acceptable by the user agent.

Thus, we need to update the logic to specify `identity` when there is no Accept Encoding header.

The problem would manifest itself in the following scenario:

1. No Accept-Encoding header is sent on a request
2. The server sends back the response encoded in an unsupported encoding (e.g. brotli/deflate)
3. Something in Cypress relies on the decompressed body (this could include `cy.intercept` or CSS, font, and image assets in Test Replay)

### Steps to test


1. Set up a server that returns brotli encoded assets when no Accept Encoding header is specified. 
2. Record a Cypress run where assets don't request an Accept Encoding. 
3. Ensure that Test Replay is still functional

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
